test/largescale/trafos/fourier_slow_test.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2014, 2015 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Large-scale tests for the Fourier transform.
+
+This test suite is intended for performance monitoring, to capture
+speed regressions.
+"""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import range
+
+# External module imports
+import pytest
+import numpy as np
+
+# ODL imports
+import odl
+from odl.util.testutils import almost_equal
+
+
+pytestmark = odl.util.skip_if_no_largescale
+
+
+# Helpers to generate data
+def _array(discr):
+    # Generate numpy vectors, real or complex or int
+    if np.issubdtype(discr.dtype, np.floating):
+        arr = np.random.rand(discr.size)
+    else:
+        arr = np.random.rand(discr.size) + 1j * np.random.rand(discr.size)
+
+    return arr.astype(discr.dtype, copy=False)
+
+
+def _element(discr):
+    return discr.element(_array(discr))
+
+
+def _vectors(discr, num=1):
+    """Create a list of arrays and elements in ``discr``.
+
+    First arrays, then vectors.
+    """
+    arrs = [_array(discr) for _ in range(num)]
+
+    # Make Discretization vectors
+    vecs = [discr.element(arr) for arr in arrs]
+    return arrs + vecs
+
+
+# Pytest fixtures
+dom_params = [odl.uniform_discr(-2, 2, 10 ** 5),
+              odl.uniform_discr([-2, -2, -2], [2, 2, 2], [200, 200, 200]),
+              odl.uniform_discr(-2, 2, 10 ** 5, dtype='complex'),
+              odl.uniform_discr([-2, -2, -2], [2, 2, 2], [200, 200, 200],
+                                dtype='complex')]
+
+dom_ids = [' {!r} '.format(dom) for dom in dom_params]
+
+
+impl_params = ['numpy']
+if odl.trafos.PYFFTW_AVAILABLE:
+    impl_params += ['pyfftw']
```

Comment:
I implemented the conditional skips with parametrized tests. However, there seems to be a bug in pytest since the global `skip_if_no_largescale` is ignored if I try to parametrize the test here in the same way as in `fourier_test.py`. The unconditional `'numpy'` case is skipped but the conditional `'pyfftw'` one runs anyway.